src/store/actions.js

```diff
@@ -11,31 +11,39 @@ const getAbiDeployedAddress = abi => {
 export default {
   // Connect to a known web3 provider
   // https://gist.github.com/bitpshr/076b164843f0414077164fe7fe3278d9#file-provider-enable-js
-  async connect({ commit, state, dispatch }) {
+  // async connect({ commit, state, dispatch }) {
+  async connect({ commit, dispatch }) {
     dispatch('initFirebase')
     let web3Provider = false
-    if (typeof window.web3 !== 'undefined') {
-      web3Provider = window.web3.currentProvider
-      try {
-        // Not quite ready yet
-        if (web3Provider.enable) await web3Provider.enable()
-        // console.log('web3Provider', web3Provider)
-        commit('SET_METAMASK', true)
-      } catch (e) {
-        console.log('e', e)
-        commit('SET_METAMASK', false)
-      }
-    } else if (!state.retried) {
-      commit('SET_RETRY', true)
-      setTimeout(() => {
-        dispatch('connect')
-      }, 1000)
-    }
-    if (state.retried && !web3Provider) {
-      web3Provider = new Web3(
-        window.web3.givenProvider || `ws://${process.env.RPC_PROVIDER}`
+    if (process.env.RPC_PROVIDER.substring(0, 3) === 'wss') {
+      web3Provider = new Web3.providers.WebsocketProvider(
+        process.env.RPC_PROVIDER
       )
+    } else if (process.env.RPC_PROVIDER.substring(0, 4) === 'http') {
+      web3Provider = new Web3.providers.HttpProvider(process.env.RPC_PROVIDER)
     }
+    // if (typeof window.web3 !== 'undefined') {
+    //   web3Provider = window.web3.currentProvider
+    //   try {
+    //     // Not quite ready yet
+    //     if (web3Provider.enable) await web3Provider.enable()
+    //     // console.log('web3Provider', web3Provider)
+    //     commit('SET_METAMASK', true)
+    //   } catch (e) {
+    //     console.log('e', e)
+    //     commit('SET_METAMASK', false)
+    //   }
+    // } else if (!state.retried) {
+    //   commit('SET_RETRY', true)
+    //   setTimeout(() => {
+    //     dispatch('connect')
+    //   }, 1000)
+    // }
+    // if (state.retried && !web3Provider) {
+    //   web3Provider = new Web3.providers.WebsocketProvider(
+    //     process.env.RPC_PROVIDER
+    //   )
```

Comment:
can we make sure to rip out all the commented blocks by the time we turn this in please? 😬